StaticAnalyzer/views/windows.py

```diff
@@ -199,53 +222,98 @@ def _binary_analysis(tools_dir, app_dir):
     bin_path = os.path.join(app_dir, bin_an_dic['bin'])
 
     # Execute strings command
-    # Works only in Mac and Linux by default.
-    args = ["strings", bin_path]
-    bin_an_dic['strings'] = escape(subprocess.check_output(args))
-    bin_an_dic['strings'] = bin_an_dic['strings'].replace("\n", "</br>")
+    bin_an_dic['strings'] = ""
+    str_list = list(strings(bin_path))
+    str_list = set(str_list)  # Make unique # pylint: disable-msg=R0204
+    str_list = [escape(s) for s in str_list]
+    bin_an_dic['strings'] = "</br>".join(str_list)
 
     # Execute binskim analysis if vm is available
-    if settings.WINDOWS_VM_IP is not None:
-        print "[INFO] Windows VM configured."
-        name = _upload_sample(bin_path)
-        bin_an_dic = __binskim(name, bin_an_dic)
-        bin_an_dic = __binscope(name, bin_an_dic)
+    if settings.CURRENT_PLATFROM != 'Windows':
+        if settings.WINDOWS_VM_IP is not None:
+            print "[INFO] Windows VM configured."
+            name = _upload_sample(bin_path)
+            bin_an_dic = __binskim(name, bin_an_dic)
+            bin_an_dic = __binscope(name, bin_an_dic)
+        else:
+            print "[INFO] Windowsc VM not configured in settings.py. Skipping Binskim and Binscope."
+            warning = {
+                "rule_id": "VM",
+                "status": "Info",
+                "desc": "VM is not configured. Please read the readme.md in MobSF/install/windows."
+            }
+            bin_an_dic['results'].append(warning)
     else:
-        print "[INFO] Windowsc VM not configured in settings.py. Skipping Binskim and Binscope."
-        warning = {
-            "rule_id": "VM",
-            "status": "Info",
-            "desc": "VM is not configured. Please read the readme.md in MobSF/install/windows."
-        }
-        bin_an_dic['results'].append(warning)
+        # TODO(Anpassen für windows lokal)
+        print "[INFO] Running lokal analysis."
+
+        global config
+        config = configparser.ConfigParser()
+        config.read('C:\\MobSF\\Config\\config.txt')
```

Comment:
Yep, but that's a general problem.

Since I wanted to reuse setup.py from the windows-vm `C:\MobSF` with the corresponding configfile, settings.py is not used.

I could also specify the options in MobSFs settings.py, but if something changes (like a download link etc.) it would have to be changed in 2 locations. Like this, all windows-related config options are in /install/windows/config.txt and it costs minimum effort to keep windows-analysis running in the vm and locally.

If it's only the problem with the permissions under `c:\`, I could move the config to an user-folder. Would that satisfy the requirements or should I work out a more advanced solution with settings.py?

I could also load settings.py on the windows-analysis-vm and use it instead of `config.txt`, but I don't know if the mash-up with is really better than the current situation. A lot of unused options will be configured for local and remote analysis.